src/main/java/graphql/schema/DataFetchingEnvironment.java

```diff
@@ -44,11 +45,9 @@
      * Returns a context argument that is set up when the {@link graphql.GraphQL#execute(String, Object)} method
      * is invoked
      *
-     * @param <T> you decide what type it is
-     *
      * @return a context object
      */
-    <T> T getContext();
+    ExecutionContext getContext();
```

Comment:
I am wary of using the ExecutionContext directly in that it exposes "execution" internals directly to the most needed of API areas.

I would rather we invented a SelectionSet / Fragments interface that we can give out that info and not more so that the "scope" of API is reduced.

By limiting the scope of API we can refactor the project better of time.

By allowing carte blanche access to ExecutionContext as it is, we create more API and lock the project into having to support it 

Can we please try to make a set of interfaces that represent what you want (even if under the covers it gets it form the executioncontext etc..)